controllers/agentcluster_controller.go

```diff
@@ -193,6 +267,9 @@ func (r *AgentClusterReconciler) createClusterDeploymentObject(agentCluster *cap
 				Name:       agentCluster.Name,
 				UID:        agentCluster.UID,
 			}},
+			Labels: map[string]string{
+				AgentClusterRefLabel: agentCluster.Name,
```

Comment:
How do we handle upgrades?
For example when a user with existing agent clusters and cluster deployments gets this version are we labeling these?

Reply:
Yes good point, I think it would be a good idea to label them (if it doesn't exist or has the wrong value)

Reply:
Modified to always ensure this label!